### PR TITLE
Added derivative of cross-tile

### PR DESCRIPTION
**Pull request by jzwar**
Also changed the "style" of the connecting interface to ensure c1 continuity between adjacent faces

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

File: splinepy/microstructure/tiles/cross_3d.py [hunk 1]
```diff
@@ -66,305 +70,407 @@ def _closing_tile(
...
+            if i_derivative == 0:
+                bound_width = boundary_width
+                fill_height = filling_height
+                inv_boundary_width = 1.0 - boundary_width
+                inv_filling_height = 1.0 - filling_height
+                center_width = 1.0 - (2 * boundary_width)
+                ctps_mid_height_top = (1.0 + filling_height) * 0.5
+                ctps_mid_height_bottom = 1.0 - ctps_mid_height_top
+                r_center = center_width * 0.5
+                v_zero = 0.0
```
Comment (j042) [inline splinepy/microstructure/tiles/cross_3d.py:102]: is `v_zero` equivalent of `kZero`?

Comment (jzwar) [reply splinepy/microstructure/tiles/cross_3d.py:102]: No, because `v_one` is sometimes also `kZero`

Comment (j042) [reply splinepy/microstructure/tiles/cross_3d.py:102]: `v_`ery interesting

Comment (jzwar) [reply splinepy/microstructure/tiles/cross_3d.py:102]: This is required to reset values in case derivatives of constants are computed

**Review by j042 — APPROVED** [review]
Great addition. How did you validate / do we have tests?

**Comment by jzwar** [comment]
> Great addition. How did you validate / do we have tests?

#392 @j042

**Pull request merged**